mass_mailing_switzerland/models/mass_mailing_contact.py

```diff
@@ -79,8 +107,30 @@ def action_update_to_mailchimp(self):
         """
         if self.env.context.get("skip_mailchimp"):
             return True
-        return super(MassMailingContact,
-                     self.with_context(lang="en_US")).action_update_to_mailchimp()
+
+        for contact_to_update in self:
+            # if previous write failed reference to mailchimp member will be lost. error 404
+            try:
+                return super(MassMailingContact,
```

Comment:
You shouldn't return here, otherwise you will exit the method at the first iteration.
```suggestion
                super(MassMailingContact,
```